tests/hyperv_api/test_it_hyperv_virtdisk.cpp

```diff
@@ -0,0 +1,191 @@
+
+    uut_t uut{};
+    hyperv::virtdisk::CreateVirtualDiskParameters params{};
+    params.path = temp_path;
+    params.size_in_bytes = 1024 * 1024 * 1024; // 1 GiB
+
+    const auto result = uut.create_virtual_disk(params);
+    ASSERT_TRUE(result);
+    ASSERT_TRUE(result.status_msg.empty());
+}
+
+TEST_F(HyperVVirtDisk_IntegrationTests, get_virtual_disk_properties)
+{
+    auto temp_path = make_tempfile_path(".vhdx");
+    std::wprintf(L"Path: %s\n", static_cast<std::filesystem::path>(temp_path).c_str());
+
+    uut_t uut{};
+    hyperv::virtdisk::CreateVirtualDiskParameters params{};
+    params.path = temp_path;
+    params.size_in_bytes = 1024 * 1024 * 1024; // 1 GiB
+
+    const auto c_result = uut.create_virtual_disk(params);
+    ASSERT_TRUE(c_result);
+    ASSERT_TRUE(c_result.status_msg.empty());
+
+    hyperv::virtdisk::VirtualDiskInfo info{};
+    const auto g_result = uut.get_virtual_disk_info(temp_path, info);
+
+    ASSERT_TRUE(info.virtual_storage_type.has_value());
+    ASSERT_TRUE(info.size.has_value());
+
+    ASSERT_STREQ(info.virtual_storage_type.value().c_str(), "vhdx");
+    ASSERT_EQ(info.size->virtual_, 1024 * 1024 * 1024);
+    ASSERT_EQ(info.size->block, 1024 * 1024);
+    ASSERT_EQ(info.size->sector, 512);
+
+    fmt::print("{}", info);
+}
+
+TEST_F(HyperVVirtDisk_IntegrationTests, resize_grow)
+{
+    auto temp_path = make_tempfile_path(".vhdx");
+    std::wprintf(L"Path: %s\n", static_cast<std::filesystem::path>(temp_path).c_str());
+
+    uut_t uut{};
+    hyperv::virtdisk::CreateVirtualDiskParameters params{};
+    params.path = temp_path;
+    params.size_in_bytes = 1024 * 1024 * 1024; // 1 GiB
+
+    const auto c_result = uut.create_virtual_disk(params);
+    ASSERT_TRUE(c_result);
+    ASSERT_TRUE(c_result.status_msg.empty());
+
+    hyperv::virtdisk::VirtualDiskInfo info{};
+    const auto g_result = uut.get_virtual_disk_info(temp_path, info);
+
+    ASSERT_TRUE(g_result);
+    ASSERT_TRUE(info.virtual_storage_type.has_value());
+    ASSERT_TRUE(info.size.has_value());
+
+    ASSERT_STREQ(info.virtual_storage_type.value().c_str(), "vhdx");
+    ASSERT_EQ(info.size->virtual_, params.size_in_bytes);
+    ASSERT_EQ(info.size->block, 1024 * 1024);
+    ASSERT_EQ(info.size->sector, 512);
+
+    fmt::print("{}", info);
+
+    const auto r_result = uut.resize_virtual_disk(temp_path, params.size_in_bytes * 2);
+    ASSERT_TRUE(r_result);
+
+    info = {};
+
+    const auto g2_result = uut.get_virtual_disk_info(temp_path, info);
+
+    ASSERT_TRUE(g2_result);
+    ASSERT_TRUE(info.virtual_storage_type.has_value());
+    ASSERT_TRUE(info.size.has_value());
+
+    ASSERT_STREQ(info.virtual_storage_type.value().c_str(), "vhdx");
+    ASSERT_EQ(info.size->virtual_, params.size_in_bytes * 2);
+    ASSERT_EQ(info.size->block, 1024 * 1024);
+    ASSERT_EQ(info.size->sector, 512);
+
+    fmt::print("{}", info);
+}
+
+TEST_F(HyperVVirtDisk_IntegrationTests, DISABLED_resize_shrink)
```

Comment:
also disabled?

Reply:
There was an issue with the shrink, but I can't recall what. Multipass does not support shrinking disks anyway, so I haven't spent much time digging further.